plugins/experimental/parallel_filesystem_operation.hpp

```diff
@@ -0,0 +1,170 @@
+#ifndef PARALLEL_FILESYSTEM_OPERATION_HPP
+#define PARALLEL_FILESYSTEM_OPERATION_HPP
+
+#include "experimental_plugin_framework.hpp"
+#include "irods_get_full_path_for_config_file.hpp"
+#include "irods_logger.hpp"
+#include "filesystem.hpp"
+#include "thread_pool.hpp"
+#include "connection_pool.hpp"
+
+#include "fmt/format.h"
+
+namespace irods::experimental::api {
+
+    namespace fs   = irods::experimental::filesystem;
+    namespace fscl = irods::experimental::filesystem::client;
+
+    class parallel_filesystem_operation : public base {
+        public:
+        parallel_filesystem_operation(const std::string& n) : base(n) {}
+        virtual ~parallel_filesystem_operation() {}
+
+        protected:
+        using entry_type    = fscl::recursive_collection_iterator::value_type;
+        using dispatch_type = dispatch_processor<fscl::recursive_collection_iterator>;
+
+        auto thread_count(const json& req)
+        {
+            if(req.contains("thread_count")) {
+                return req.at("thread_count").get<int>();
+            }
+
+            std::string path{};
+            if (auto e = irods::get_full_path_for_config_file("server_config.json", path); e.ok()) {
+                json cfg{};
+
+                { std::ifstream f{path}; f >> cfg; }
+
+                auto psc = cfg.at(irods::CFG_PLUGIN_CONFIGURATION_KW);
+
+                if(psc.contains("api") && psc.at("api").contains(instance_name_)) {
+                    auto in = psc.at("api").at(instance_name_);
+                    if(in.contains("thread_count")) {
+                        return in.at("thread_count").get<int>();
+                    }
+                }
+            }
+
+            return 4;
```

Comment:
comment about why 4 is a default here?

Reply:
It is the default for the delayed execution server and a polite number at around 10% the core/thread count of a typical server.  I should be using a constant here but was not sure if we wanted those two to be the same or name another thing locally.